### PR TITLE
FALCON-1957 Documentation on credential provider alias for passwords in startup properties

### PR DESCRIPTION
**Pull request by yzheng-hortonworks**
Also restructured the overview of Falcon security features.

### CODE REVIEW

File: docs/src/site/twiki/Security.twiki [hunk 1]
```diff
@@ -2,8 +2,30 @@
...
+In addition to specifying properties in plain text, we provide the user an option to use credential provider alias in the property file.
+
+Take SMTP password for example. The user can store the password in a
+[[http://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-common/CommandsManual.html#credential][Hadoop credential provider]]
+with the alias name _SMTPPassword_. In startup.properties where SMTP password is needed, the user can refer to its
+alias name _SMTPPassword_ instead of providing the real password.
+
+The alias property to be resolved through Hadoop credential provider should have the format:
+_credential.provider.alias.for.[property-key]_. For example,
+_credential.provider.alias.for.falcon.email.smtp.password=SMTPPassword_ for SMTP password.
```
Comment (vrangan) [inline docs/src/site/twiki/Security.twiki]: Shouldn't this read

credential.provider.alias.for.falcon.email.smtp.password=Alias for SMTPPassword?

Comment (yzheng-hortonworks) [inline docs/src/site/twiki/Security.twiki]: SMTPPassword is the alias (see previous paragraph: "...with the alias name _SMTPPassword_.")

Comment (vrangan) [inline docs/src/site/twiki/Security.twiki]: That is confusing  - may be you can rename the alias name to say SMTPPasswordAlias and refer it here?

Comment (yzheng-hortonworks) [inline docs/src/site/twiki/Security.twiki]: Good point, @vrangan . I added "Alias" to the alias name.

**Comment by vrangan** [comment]
+1

**Comment by bvellanki** [comment]
+1 will commit soon

**Pull request closed without merging**